components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.v1/src/main/java/org/wso2/carbon/identity/api/server/userstore/v1/impl/UserstoresApiServiceImpl.java

```diff
@@ -41,9 +42,21 @@
     @Autowired
     private ServerUserStoreService serverUserStoreService;
 
+
+    private static boolean isAvailableUserStoreTypes(List<AvailableUserStoreClassesRes> userStoreList, String typeID) {
+        for (AvailableUserStoreClassesRes userStore : userStoreList) {
+            if (userStore.getTypeId().equals(typeID)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
     @Override
     public Response addUserStore(UserStoreReq userStoreReq) {
-
+        if (!isAvailableUserStoreTypes(serverUserStoreService.getAvailableUserStoreTypes(), userStoreReq.getTypeId())) {
+            return Response.status(Response.Status.BAD_REQUEST).build();
```

Comment:
let's give an error message in the response

Reply:
Fixed in commit f5ea11e14fef1323d232a58286f4ebb697559da1